band_tracker/db/dal_update.py

```diff
@@ -170,6 +182,46 @@ def _build_db_socials(
         )
         return socials_db
 
+    async def _get_new_genres(
+        self, session: AsyncSession, artist_id: UUID, genres: list[str]
+    ) -> list[GenreDB]:
+        artist_genre_names = (
+            select(GenreDB.name)
+            .join(ArtistGenreDB, GenreDB.id == ArtistGenreDB.genre_id)
+            .filter(ArtistGenreDB.artist_id == artist_id)
+        )
+        execution_result = await session.execute(artist_genre_names)
+        current_genre_names = [name[0] for name in execution_result]
+
+        new_genre_names = [
+            genre for genre in genres if genre not in current_genre_names
+        ]
+        new_db_genres = await self._get_db_genres(session, new_genre_names)
+
+        return new_db_genres
+
+    async def _get_genre(self, session: AsyncSession, name: str) -> GenreDB | None:
+        genre_query = select(GenreDB).where(GenreDB.name == name)
+        scalar = await session.scalars(genre_query)
+        genre_db = scalar.first()
+        return genre_db
+
+    def _add_genre(self, name: str) -> GenreDB:
+        genre_db = GenreDB(name=name)
+        return genre_db
+
+    async def _get_db_genres(
+        self, session: AsyncSession, genres: list[str]
+    ) -> list[GenreDB]:
+        result = []
+        for genre in genres:
+            genre_db = await self._get_genre(session, genre)
```

Comment:
probably should use 1 single query

Reply:
get all genres from a list with 1 query (using .filter and in_), and work with that to avoid multiple i/o